webapp/src/main/resources/application.sql.yml

```diff
@@ -9,6 +9,20 @@ sql:
       from district
       where natural_id in (:naturalIds)
 
+  group:
+    findAllByQuery: >-
+      select sg.name as group_name,
+        sc.name as school_name,
+        su.code as subject_code,
+        sg.school_year
+      from student_group sg
+        join school sc on sc.id = sg.school_id
+        left join subject su on su.id = sg.subject_id
+      where sg.school_id = :school_id
+        and (sg.subject_id is null or sg.subject_id in (:subject_ids))
```

Comment:
subject_id of `null` is inferred to include all subjects, so we always include `null` subject_ids in the query regardless of the subject_ids passed in.  We do this in Reporting as well for permission checking.